components/custom_status/custom_status_modal.tsx

```diff
@@ -177,12 +177,12 @@ const CustomStatusModal: React.FC<Props> = (props: Props) => {
         setExpiry(defaultDuration);
     };
 
-    let disableSetStatus = (currentCustomStatus.text === text && currentCustomStatus.emoji === emoji) ||
+    let disableSetStatus = (currentCustomStatus?.text === text && currentCustomStatus.emoji === emoji) ||
         (text === '' && emoji === '') || (text.length > Constants.CUSTOM_STATUS_TEXT_CHARACTER_LIMIT);
 
     disableSetStatus = Boolean(disableSetStatus &&
-        ((expiry !== CustomStatusDuration.DATE_AND_TIME && currentCustomStatus.duration === expiry) ||
-            (expiry === CustomStatusDuration.DATE_AND_TIME && currentCustomStatus.expires_at && customExpiryTime.getTime() === new Date(currentCustomStatus.expires_at).getTime())));
+        ((expiry !== CustomStatusDuration.DATE_AND_TIME && currentCustomStatus?.duration === expiry) ||
+            (expiry === CustomStatusDuration.DATE_AND_TIME && currentCustomStatus?.expires_at && customExpiryTime.getTime() === new Date(currentCustomStatus.expires_at).getTime())));
```

Comment:
nit - if possible, refactor this to multiple boolean variables to facilitate ease of readability